app/controllers/results_controller.rb

```diff
@@ -320,7 +320,7 @@ def next_grouping
         next_pr = assigned_prs.where('peer_reviews.id < ?', result.peer_review_id).last
       end
       next_result = Result.find(next_pr.result_id)
-      redirect_to action: 'edit', id: next_result.id
+      render json: { next_result: next_result, next_grouping: next_pr }
```

Comment:
It would be great to not duplicate the code here. (Set `next_grouping` and `next_result` in both branches, and then keep the `render` call at the bottom of the file.)